MekHQ/src/mekhq/campaign/LogEntry.java

```diff
@@ -35,116 +35,124 @@
-                } else if (wn2.getNodeName().equalsIgnoreCase("date")) {
-                    retVal.date = DATE_FORMAT.parse(wn2.getTextContent().trim());
-                }
-            }
-        } catch (Exception ex) {
-            // Doh!
-            MekHQ.getLogger().log(LogEntry.class, METHOD_NAME, ex);
-        }
-        
-        return retVal;
-    }
-    
-    @Override
-    public String toString() {
-        StringBuilder sb = new StringBuilder();
-        if(null != date) {
-            sb.append("[").append(DATE_FORMAT.format(date)).append("] ");
-        }
-        sb.append(desc);
-        return sb.toString();
-    }
-    
-    @Override
-    public LogEntry clone() {
-        return new LogEntry(getDate(), getDesc(), getType());
-    }
-}
\ No newline at end of file
+public class LogEntry implements Cloneable, MekHqXmlSerializable {
+
+	private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss"); //$NON-NLS-1$
+	// LATER SimpleDateFormat is not thread-safe: use throw-away instances or switch to java.time
+	// private static final DateTimeFormatter DTF = DateTimeFormatter.ofPattern("", Locale.ENGLISH);
+
+	public LogEntry(Date date, String desc) {
+		this(date, desc, null);
+	}
+	
+	public LogEntry(Date date, String desc, String type) {
+		this.date = date;
+		this.desc = desc != null ? desc : ""; //$NON-NLS-1$
+		this.type = type;
+	}
+
+	private Date date;
+	private String desc; // non-null
+	private String type;
+
+	public Date getDate() {
+		return date;
+	}
+	
+	public void setDate(Date date) {
+		this.date = date;
+	}
+	
+	public String getDesc() {
+		return desc;
+	}
+	
+	public void setDesc(String desc) {
+		this.desc = desc != null ? desc : ""; //$NON-NLS-1$
+	}
+	
+	public String getType() {
+		return type;
+	}
+	
+	public void setType(String type) {
+		this.type = type;
+	}
+
+	@Override
+	public void writeToXml(PrintWriter pw, int indent) {
+		StringBuilder sb = new StringBuilder();
+		sb.append(MekHqXmlUtil.indentStr(indent)).append("<logEntry>"); //$NON-NLS-1$
+		if (date != null)    sb.append("<date>").append(DATE_FORMAT.format(date)).append("</date>"); //$NON-NLS-1$ //$NON-NLS-2$
+		assert desc != null; sb.append("<desc>").append(MekHqXmlUtil.escape(desc)).append("</desc>"); //$NON-NLS-1$ //$NON-NLS-2$
+		if (type != null)    sb.append("<type>").append(MekHqXmlUtil.escape(type)).append("</type>");  //$NON-NLS-1$//$NON-NLS-2$
+		sb.append("</logEntry>"); //$NON-NLS-1$
+		pw.println(sb.toString());
+	}
+
+	public static LogEntry generateInstanceFromXML(Node wn) {
+		final String METHOD_NAME = "generateInstanceFromXML(Node)"; //$NON-NLS-1$
+
+		Date   date = null;
+		String desc = null;
+		String type = null;
+
+		try {
+			NodeList nl = wn.getChildNodes();
+			for (int x = 0; x < nl.getLength(); x++) {
+				Node   node = nl.item(x);
+				String nname = node.getNodeName();
+				if (nname.equalsIgnoreCase("desc")) { //$NON-NLS-1$
```

Comment:
Minor nit: XML element names are case sensitive, so the `equalsIgnoreCase` could be just an `equals`. (I realize the rest of MekHQ probably uses `equalsIgnoreCase`)

Reply:
Good catch!

I've seen a mix of `equals(s)`, `equalsIgnoreCase(s)` and `contains(s)` around the codebase, but didn't think of double checking the very class I was modifying :-)